### ISSUE TITLE
Social.localUser.Authenticate always returns false even though Google Play login Succeeds

### ISSUE THREAD

**Issue by Spud6703** [playgameservices/play-games-plugin-for-unity#2620]
When I try to authenticate the user with the following code it always returns success = false

`
Social.localUser.Authenticate((bool success) =>
{
    // handle success or failure
    if (success)
    {
        Debug.Log("Authentication successful");
    }
    else
    {
        Debug.Log("Authentication failed");
    }
});
`

but when I look at the logcat, it says "GamesNativeSDK: Successfully connected to Google Play." This happens after Social.localUser.Authenticate has returned false. I'm using Unity 2019.1.7f1.

(btw, when I paste the code into the block above and Preview it, it's all on the same line - why ?)

**Comment by Spud6703** [comment]
If I poll for Social.localUser.authenticated in a Monobehaviour.Update() function then it eventually returns true and I can retrieve my Google Play login details. So everything is working with the Google Play plugin .... except the callback for Social.localUser.Authenticate returns false way before Google Play has even had a chance to try logging in

**Comment by Spud6703** [comment]
SUCCESS!! Okay I managed to fix it by linking a Web App to my Android App in the Google Play Game Services console and copying the Client Id into the Web App Client ID (optional) box in the Unity Google Play/Setup thing. This was all part of trying to set up GameSparks to authenticate with Google Play but after making these changes I started getting a privacy policy notice popping up at login and then Social.localUser.Authenticate returned true at the correct place. I'll close this issue then!
